packages/backend-core/src/redis/redis.ts

```diff
@@ -274,13 +279,44 @@ class RedisWrapper {
     }
   }
 
+  async bulkStore(
+    data: Record<string, any>,
+    expirySeconds: number | null = null
+  ) {
+    const client = this.getClient()
+
+    const dataToStore = Object.entries(data).reduce((acc, [key, value]) => {
+      acc[addDbPrefix(this._db, key)] =
+        typeof value === "object" ? JSON.stringify(value) : value
+      return acc
+    }, {} as Record<string, any>)
+
+    const luaScript = `
+        for i, key in ipairs(KEYS) do
+            redis.call('MSET', key, ARGV[i])
+            ${
+              expirySeconds !== null
+                ? `redis.call('EXPIRE', key, ARGV[#ARGV])`
+                : ""
+            }
+        end
+        `
+    const keys = Object.keys(dataToStore)
+    const values = Object.values(dataToStore)
+    if (expirySeconds !== null) {
+      values.push(expirySeconds)
+    }
+
+    await client.eval(luaScript, keys.length, ...keys, ...values)
+  }
```

Comment:
How come we can't use `mset` directly? https://redis.github.io/ioredis/classes/Redis.html#mset

Slightly concerned that Redis mark the use of `eval` as `@slow` in their docs: https://redis.io/commands/eval/

Reply:
We can, but there is no way to add the expiration in bulk without eval.
We could add the keys first and then modify the expiration (as we do in the actual `store`), but probably it's best to have it as an atomic operation

Reply:
I will play with the pipeline option. It's not transactional, but at least it will ensure atomicity

Reply:
`multi` might be more appropriate for this use-case: https://redis.io/commands/multi/

I'm not 100% sure, though, and reading a bit more around scripts they do guarantee atomicity provided you only access the keys you say you will. I suspect the code using `multi` might look a bit nicer, but ultimately leave it up to you to decide.